training/splitting_dataset.py

```diff
@@ -0,0 +1,132 @@
+        for book_name in books_per_year[year]:
+            book_df = dataframe.loc[dataframe["book"] == book_name]
+            included_rows_num = min(remaining_rows_num, book_df.shape[0])
+            filtered_dataframe = pd.concat(
+                [filtered_dataframe, book_df[:included_rows_num]]
+            )
+            remaining_rows_num -= included_rows_num
+            if remaining_rows_num == 0:
+                return filtered_dataframe
+    return filtered_dataframe
+
+
+def _get_chronological_split_results(
+    dataframe: DataFrameType, test_size: float = _DEFAULT_TEST_SIZE
+) -> Tuple[DataFrameType, DataFrameType, DataFrameType, DataFrameType]:
+    """
+    Splits dataframe into training and testing sets chronologically.\n
+    :param dataframe: Processed dataframe.
+    :param test_size: Fraction of the dataset to include in test split. It should be a float number between 0.0 and 1.0.
+    :return: Two dataframes (first one for training and second one for testing) and two columns in the same order.
+    """
+    X_train = _get_dataframe_with_oldest_books(dataframe, test_size)
+    if not X_train.empty:
+        X_test = dataframe.drop(X_train.index)
+        y_train = X_train.pop("level")
+        y_test = X_test.pop("level")
+        return X_train, X_test, y_train, y_test
+    else:
+        print(
+            "_get_chronological_split_results: No books were found in dataframe. Dataframe will be split randomly."
+        )
+        return _get_random_split_results(dataframe)
+
+
+def _get_random_split_results(
+    dataframe: DataFrameType, test_size: float = _DEFAULT_TEST_SIZE
+) -> Tuple[DataFrameType, DataFrameType, DataFrameType, DataFrameType]:
+    """
+    Splits dataframe into training and testing sets randomly.\n
+    :param dataframe: Processed dataframe.
+    :param test_size: Fraction of the dataset to include in test split. It should be a float number between 0.0 and 1.0.
+    :return: Two dataframes (first one for training and second one for testing) and two columns in the same order.
+    """
+    X, y = dataframe.drop("level", axis="columns"), dataframe["level"]
+    return train_test_split(X, y, test_size=test_size, random_state=0, shuffle=True)
+
+
+def split_dataframe(
+    dataframe: DataFrameType,
+    test_size: float = _DEFAULT_TEST_SIZE,
+    random_split: bool = False,
+) -> Tuple[DataFrameType, DataFrameType, DataFrameType, DataFrameType]:
+    """
+    Splits dataframe to dataframes allowing training and testing of ML models. Requires "level" column to be
+    present in initial dataframe.\n
+    :param dataframe: Processed dataframe.
+    :param test_size: Fraction of the dataset to include in test split. It should be a float number between 0.0 and 1.0.
+    :param random_split: If False, splits dataframe chronologically so that training set contains rows with
+    the oldest books from initial dataframe and testing set contains the newest. Otherwise, the dataframe is
```

Comment:
Niestety koncepcja jest inna (przynajmniej z tego co ja zrozumiałam)
co najmniej X% potworów z najnowszych książek - i tu nie jestem pewna czy to jest min na test, train czy oba
Pewnie obstawiłabym oba

Reply:
Przydatne może okazać się: stratify
Niestety wszędzie zazwyczaj są wpisy o k-fold, ale może się przyda:
https://scikit-learn.org/stable/modules/cross_validation.html#stratification